### PR TITLE
Updates to make the training process a bit smoother

### PR DESCRIPTION
**Pull request by savagej**
- Increased number of epochs to `14` to match documented default, looks like that gives best bang for buck
- Added ability to train with GPU
- Model saving tidy up
  - Added option to turn off logging of the model to MLflow
  - Fixed issues with the `save_model` arg, MLProject file doesn't do bool parameters
  - Removed local saving of the model altogether
- Fixed bug where accuracy 16x too high
- Removed awkward use of `None` in parameters, just using strings as boolean approximation

### CODE REVIEW

File: src/bert-classifier/Dockerfile.base [hunk 1]
```diff
@@ -2,13 +2,16 @@
...
 COPY requirements.txt /tmp/
 
 RUN mkdir -p /usr/share/man/man1/ && \
     apt-get update && apt-get install --no-install-recommends -y git default-jdk wget jq uuid-runtime && \
     apt-get clean -y -qq && apt-get autoremove -y -qq && \
     rm -rf /var/lib/apt/lists/* && \
     git clone https://github.com/pytorch/serve.git && \
-    cd serve && python ./ts_scripts/install_dependencies.py && \
+    cd serve && \
+    if [ -z "$CUDA_VERSION" ]; then python ts_scripts/install_dependencies.py; else python ts_scripts/install_dependencies.py --cuda "$CUDA_VERSION"; fi && \
```
Comment (savagej) [inline src/bert-classifier/Dockerfile.base:15]: Delighted we used this `install_dependencies` script now!! Got cuda installation essentially for free!

https://github.com/pytorch/serve/blob/master/docker/Dockerfile.dev#L56

**Review by michelav — APPROVED** [review]
Cool!

**Comment by savagej** [comment]
http://35.185.111.8:5000/#/metric/train_acc?runs=[%221f8e440566d4494a8eb20c5d22e940f0%22]&experiment=0&plot_metric_keys=[%22train_acc%22,%22test_acc%22,%22val_acc%22]&plot_layout={%22autosize%22:true,%22xaxis%22:{},%22yaxis%22:{}}&x_axis=relative&y_axis_scale=linear&line_smoothness=1&show_point=false&deselected_curves=[]&last_linear_y_axis_range=[]
